### PR TITLE
Remove unused isTestRule for ObjC builds

### PR DESCRIPTION
**Pull request by keith**
This was likely left over from before rules_apple was split out. This
was never set to true.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/rules/objc/CompilationSupport.java [hunk 1]
```diff
@@ -799,12 +782,6 @@ public Builder doNotUsePch() {
       return this;
     }
 
-    /** Indicates that this CompilationSupport is for use in a test rule. */
-    public Builder setIsTestRule() {
```
Comment (keith) [inline src/main/java/com/google/devtools/build/lib/rules/objc/CompilationSupport.java:803]: The key to all this being dead was this not being called

**Comment by lberki** [comment]
Hey, this is nice! I'll let @allevato do the actual review, though, because I'm at best a recurrent visitor in our Objective-C rules.

File: src/main/java/com/google/devtools/build/lib/rules/objc/CompilationSupport.java [hunk 2]
```diff
@@ -894,7 +870,7 @@ public InstrumentedFilesInfo getInstrumentedFilesProvider(ImmutableList<Artifact
         NestedSetBuilder.<Artifact>emptySet(Order.STABLE_ORDER),
         // The COVERAGE_GCOV_PATH environment variable is added in TestSupport#getExtraProviders()
         NestedSetBuilder.<Pair<String, String>>emptySet(Order.COMPILE_ORDER),
-        !isTestRule,
+        true,
```
Comment (allevato) [inline src/main/java/com/google/devtools/build/lib/rules/objc/CompilationSupport.java:873]: Can you precede this argument with a comment like the one below that indicates the formal argument name?

**Review by allevato — APPROVED** [review]
This looks fine if it's not being used, but we should have @googlewalt or @trybka take a look at how the removal of this feature would affect our version of crosstool before it gets pulled in.

File: src/main/java/com/google/devtools/build/lib/rules/objc/CompilationSupport.java (hunk 2 above)
Comment (keith) [reply src/main/java/com/google/devtools/build/lib/rules/objc/CompilationSupport.java:873]: Done!

**Comment by googlewalt** [comment]
> This looks fine if it's not being used, but we should have @googlewalt or @trybka take a look at how the removal of this feature would affect our version of crosstool before it gets pulled in.

I filed a bug internally to look into this.  Our internal crosstool still references this, so we need to determine whether the functionality is still useful but broken, or obsolete.

**Comment by keith** [comment]
Potentially related: https://github.com/bazelbuild/bazel/issues/12176

**Comment by googlewalt** [comment]
LGTM.  I will try to merge this.

**Review by googlewalt — APPROVED** [review]

**Pull request closed without merging**